### PR TITLE
Enforce a minimum version for Kusto and SQL Service dotnet tools.

### LINKED ISSUE #2681: Can't connect to Kusto
All good yesterday. Today I open an notebook and can't connect to Kusto. Tried this on 2 different machines.  Wondering what changed. Any pointers? 

`Error: System.InvalidOperationException: MicrosoftKustoServiceLayer failed to start properly. Exit code: 0.
 ---> StreamJsonRpc.ConnectionLostException: The JSON-RPC connection with the remote party was lost before the request could complete.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync(JsonRpcRequest request, Type expectedResultType, CancellationToken cancellationToken)
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.DotNet.Interactive.SqlServer.ToolsServiceClient.ConnectAsync(ConnectParams connectParams)
   --- End of inner exception stack trace ---
   at Microsoft.DotNet.Interactive.SqlServer.ToolsServiceClient.ConnectAsync(ConnectParams connectParams)
   at Microsoft.DotNet.Interactive.Kql.ToolsServiceClientExtensions.ConnectAsync(ToolsServiceClient serviceClient, Uri ownerUri, KqlConnectionDetails kqlDetails)
   at Microsoft.DotNet.Interactive.Kql.MsKqlKernel.ConnectAsync()
   at Microsoft.DotNet.Interactive.Kql.KqlKernelConnector.CreateKernelAsync(String kernelName)
   at Microsoft.DotNet.Interactive.Kql.ConnectKqlCommand.ConnectKernelAsync(KernelInvocationContext context, InvocationContext commandLineContext)
   at Microsoft.DotNet.Interactive.CompositeKernel.<>c__DisplayClass26_0.<<AddKernelConnector>b__0>d.MoveNext() in D:\a\_work\1\s\src\Microsoft.DotNet.Interactive\CompositeKernel.cs:line 329`

### PR DESCRIPTION
**Pull request by corivera**
A while back, I checked in a version bump for the dotnet tools we install as part of starting up the Kusto and SQL kernels. However, I forgot to add a mechanism for enforcing a minimum version of those tools, which is a problem because we changed how we launch the tool in the Kusto kernel by adding an extra argument, which is not supported in the previous version of the tool. This change checks if the package is installed, and then whether it needs to be updated if it's already present. 

I verified the following scenarios:
1. Old 1.0.0 tool installed for Kusto and SQL
2. Current 1.1.0 tool installed for both
3. No version of either tool installed

I also made sure to keep the changes contained to the KQL and SqlServer projects, so we only need to update those nuget packages.

Fixes #2681

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Interactive.Utility;
+
 namespace Microsoft.DotNet.Interactive.SqlServer;
 
+public class DotnetToolInfo {
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs:12]: It's best to keep this internal.

File: src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+    public string PackageId { get; set; }
+    public string PackageVersion { get; set; }
+    public string CommandName { get; set; }
+
+}
+
 public static class Utils
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs:19]: Can this be made internal?

File: src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs (hunk 1 above)
Comment (corivera) [reply src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs]: Done

File: src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs (hunk 2 above)
Comment (corivera) [reply src/Microsoft.DotNet.Interactive.SqlServer/Utils.cs]: Done

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**